src/commands/aksInspektorGadget/aksInspektorGadget.ts

```diff
@@ -0,0 +1,175 @@
+    target: any
+): Promise<void> {
+    await gadgetDeployUndeploy(_context, target, GadgetCommand.Undeploy);
+}
+
+async function gadgetDeployUndeploy(
+    _context: IActionContext,
+    target: any,
+    gadget: GadgetCommand
+): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return undefined;
+    }
+
+    if (!cloudExplorer.available) {
+        vscode.window.showWarningMessage(`Cloud explorer is unavailable.`);
+        return undefined;
+    }
+
+    if (!clusterExplorer.available) {
+        vscode.window.showWarningMessage(`Cluster explorer is unavailable.`);
+        return undefined;
+    }
+
+    const clusterInfo = await getKubernetesClusterInfo(target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return undefined;
+    }
+
+    await prepareInspektorGadgetInstall(clusterInfo.result, gadget, kubectl);
+}
+
+async function prepareInspektorGadgetInstall(
+    clusterInfo: KubernetesClusterInfo,
+    gadget: GadgetCommand,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>
+): Promise<void> {
+    const clustername = clusterInfo.name;
+    const kubeconfig = clusterInfo.kubeconfigYaml;
+
+    switch (gadget) {
+        case GadgetCommand.Deploy:
+            await deployGadget(clustername, kubeconfig, kubectl);
+            return;
+        case GadgetCommand.Undeploy:
+            const answer = await vscode.window.showInformationMessage(`Do you want to undeploy gadget in ${clustername}?`, "Yes", "No");
+            if (answer === "Yes") {
+                await unDeployGadget(clustername, kubeconfig, kubectl);
+            }
+            return;
+    }
+}
+
+async function deployGadget(
+    clustername: string,
+    clusterConfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>) {
+    const command = "deploy";
+
+    return await runKubectlGadgetCommands(clustername, command, clusterConfig, kubectl);
+}
+
+async function unDeployGadget(
+    clustername: string,
+    clusterConfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>) {
+    const command = "undeploy";
+
+    return await runKubectlGadgetCommands(clustername, command, clusterConfig, kubectl);
+}
+
+async function runKubectlGadgetCommands(
+    clustername: string,
+    command: string,
+    clusterConfig: string,
+    kubectl: k8s.APIAvailable<k8s.KubectlV1>) {
+
+    const kubectlGadgetPath = await getKubectlGadgetBinaryPath();
+    let kubetlGadgetBinaryPath = "";
+
+    if (failed(kubectlGadgetPath)) {
+        vscode.window.showWarningMessage(`Gadget path is not found ${kubectlGadgetPath.error}`);
+        return;
+    }
+
+    kubetlGadgetBinaryPath = kubectlGadgetPath.result;
```

Comment:
nit: maybe combine the declaration and assignment of `kubectlGadgetBinaryPath`, and make it a `const`? Could also move it closer to where it's used... Also, if we don't need the `console.log` on line 131, we could probably get away without this identifier, and just use `kubectlGadgetPath.result` instead?